src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -169,6 +169,12 @@ export class SavedSearchEmbeddable
     return true;
   }
 
+  private isTextBasedQuery = (savedSearch: SavedSearch): boolean => {
+    const query = savedSearch.searchSource.getField('query');
+    const recordRawType = getRawRecordType(query);
+    return recordRawType === RecordRawType.PLAIN;
```

Comment:
`isPlainRecord` would also be an option here 
https://github.com/elastic/kibana/blob/dec52ef09d27645af81de4f8987a5a31d74b50b7/src/plugins/discover/public/application/main/utils/get_raw_record_type.ts#L29